mantidproject/mantid
# Add and update settings, paths, and documentation for the tomography GUI

Description of work: Introduces several parameters which are now set with sensible default values, and available in a new "system" tab.  Simplifies the "setup" path and "hides" the too many underlying details in a "system" tab. After a next demo/practical session we may simplify it more.

**To test:**
- Check code changes, and  that documentation changes make sense
- Updated unit tests of the interface, and some new cases in the
  system test of the tomography scripts. All should pass.
- This GUI is in MantidPlot, open  "Interfaces -> Diffraction -> Tomographic Reconstruction"
- Check the login: in the "setup" tab try to give empty user, empty password, bad password,
  and see that the behavior is intuitive. Close and open again the
  interface. The username should be remembered (but the password should
  not).
- We can see the options in action if we start a reconstruction job on
  the cluster (for that you need my help/login credentials).
- To check that the job setups are correct you can use the "local"
  execution (tab Run, select "Compute Resource" Local). Then, with log
  level "information" in the Results Log window you should be able to
  see the details of the remote job submission algorithm.

For example, the `SubmitRemoteJob` algorithm should run with the options set to something like:

```
--tool=tomopy --algorithm=gridrec --num-iter=5  --input-path=/some/path/ --input-path-flat=/some/other/path --input-path-dark=/one/more/path  --output="/work/imateee/dataa/RB000000/sampleA/processedf/out_reconstruction_TomoPy_gridrec_20\
16March18_163221_191790000" --median-filter-size=3 --cor=255.000000 --rotation=0 --max-angle=360.000000 --circular-mask=0.940000 --out-img-format=png
```

(this can be checked in the table of jobs and also in the logs if you set the level to information).

Only after you open a stack (in the "ROI" tab) you should see additional parameters in the job submission string. For example:

```
--region-of-interest='[0.000000, 0.000000, 511.000000, 511.000000]'
```

or

```
--air-region='[13.000000, 16.000000, 236.000000, 239.000000]'
```
- Another point to test would be:
  - Go to "Setup" tab, in the directories untick flats and darks
  - Then submit a job, see that the options `--input-path-flat` and `--input-path-dark` are not given.
  - The status of these two checkboxes (flats and darks) should be remembered between sessions, as all the other fields.
  - Go to the "system" tab. Edit a few fields with any values, and check that the "Reset all" (bottom right) works as you would expect.
- Try a local submission and see that the jobs fail ("Exit" status), because most likely you don't have the required tools installed (tomopy, astra).
- For remote submissions (you'll need my login) we should see the
  output files generated on the remote file system in the right
  locations

Fixes #15665.

Release note: https://github.com/mantidproject/mantid/commit/2f90077eab7ff4501273998bebe54a44a984d9d9

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
